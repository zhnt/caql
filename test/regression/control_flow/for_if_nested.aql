// For loops nested inside if statements
let mode = "count_evens"
let count = 0

if mode == "count_evens" {
    print("Counting even numbers")
    for i = 1, 20, 1 {
        if i % 2 == 0 {
            count = count + 1
        }
    }
    print("Even count: " + string(count))
} elif mode == "count_odds" {
    print("Counting odd numbers")
    for i = 1, 20, 1 {
        if i % 2 == 1 {
            count = count + 1
        }
    }
    print("Odd count: " + string(count))
} else {
    print("Unknown mode")
    for i = 1, 5, 1 {
        print("Default: " + string(i))
    }
}

print("Nested test complete")
