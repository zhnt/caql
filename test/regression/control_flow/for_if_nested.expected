Counting even numbers
Even count: 9
Nested test complete
